### ISSUE TITLE
Find out why doT is so much faster

### ISSUE THREAD

**Issue by TimothyGu** [mde/ejs#49]
This is more of a WIP for myself.
- [x] benchmark its escape function
- [x] ~~See if https://github.com/olado/doT/blob/master/doT.js#L48-L51 trick can be applied to EJS.~~ Nope.
- [ ] Hire a [wizard](http://browserify.org/images/wizard.png) to make EJS run as fast as doT.

**WARNING:** to anyone out there thinking tj/ejs is unreadable, please do not try on this task.

**Comment by TimothyGu** [comment]
Escape function ported in #50.

Now templates generated with `!_with` mode is about 55% as fast as doT's.

**Comment by TimothyGu** [comment]
Also note that even doT *compilation* is faster. By a lot. (By *a lot* I mean *almost 500%*.)

**Comment by TimothyGu** [comment]
OK, after #50 and #56, `!_with` is about 60% faster than doT (yes!!!!), while using `_with` turns out to be marginally slower (<5%).

(Measured using `client` mode and copied the functions into a separate file. For some reason when using all functions in the same script as compilation makes EJS slower while making doT faster.)

Compilation is marginally faster as well.

**Comment by bentinata** [comment]
I'm moving from Swig, as it unmaintained, and inbetween EJS and doT. I know you might be biased, but, what's the pros and cons of EJS and doT? Also, what should I do to make EJS better?

**Comment by TimothyGu** [comment]
@bentinata doT has an inherently more complex syntax, and thus a steeper learning curve is required. EJS is faster to learn, but once you get used to the doT "concise" syntax you might find a lot of `<% if (...) -%>` redundant.

**Comment by mde** [comment]
This issue has been closed as outdated. 

If you believe this issue is still relevant, please:
1. Open a new issue with current context
2. Reference this issue number (#49)
3. Include details about your use case and EJS version

Thank you for your contribution!
